packages/playwright-core/src/server/browserContext.ts

```diff
@@ -274,6 +276,10 @@ export abstract class BrowserContext extends SdkObject {
     await Promise.all(Array.from(this._downloads).map(download => download.artifact.deleteOnContextClose()));
   }
 
+  private async _deleteAllTempDirs(): Promise<void> {
+    await Promise.all(this._tempDirs.map(async dir => await fs.promises.unlink(dir).catch(e => {})));
```

Comment:
this sounds like we want to have it in the `packages/playwright-core/src/utils/processLauncher.ts>gracefullyCloseSet` as well.

Reply:
We careate the temp dirs in artifacts dir which is already managed and deleted by the launcher. This code juste does the cleanup earlier (we don't want to keep large file for longer than necessary and may even make the cleanup on navigation/page close in the future).